### PR TITLE
`CryptoCreate` Refactoring

### PR DESCRIPTION
**Pull request by Daniel-K-Ivanov**
This PR modifies the `CryptoCreateTransitionLogic` to use the new `Account` model and `AccountStore` instead of Hedera Ledger. It introduces the possibility of a given `account` model to transfer hbars to another account without using the Ledger.

**Checklist**
- [X] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by joan41868** [comment]
### Questions:
There is a pattern established in the `TokenCreate` OP, which decouples the logic. Should we follow up and use the same pattern here?

**Comment by Daniel-K-Ivanov** [comment]
Making this PR as draft as there is a problem with the externalisation of the Hbar transfers. We cannot simply disregard the hbar transfers from the ledger and use the explicit transfers since they include the node/service fees

File: hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoCreateTransitionLogic.java [hunk 1]
```diff
@@ -64,65 +65,76 @@
...
+
+		/* --- Load the model objects --- */
+		final var sponsor = accountStore.loadAccount(sponsorId);
+		validateFalse(sponsor.isSmartContract(), ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
+
+		/* --- Do the business logic --- */
+		var createdIdGrpc = ids.newAccountId(sponsor.getId().asGrpcAccount());
+		final var created = Account.createFromGrpc(Id.fromGrpcAccount(createdIdGrpc), op, txnCtx.consensusTime().getEpochSecond());
+		
+		validateTrue(sponsor.getBalance() > op.getInitialBalance(), INSUFFICIENT_PAYER_BALANCE);
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoCreateTransitionLogic.java:114]: Should this check be `>` or `>=`?

File: hedera-node/src/main/java/com/hedera/services/utils/GrpcTransfersBuilder.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ */
+
+
+import com.hederahashgraph.api.proto.java.AccountAmount;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.TransferList;
+
+import static com.hedera.services.ledger.HederaLedger.ACCOUNT_ID_COMPARATOR;
+
+public class GrpcTransfersBuilder {
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/main/java/com/hedera/services/utils/GrpcTransfersBuilder.java:32]: This can/should be reverted

**Review by Daniel-K-Ivanov — COMMENTED** [review]
We will need to revert the changes in `BasicTransactionContext.java`, `EntityCreator.java`, `ExpiringCreations.java` and `NoopExpiringCreations.java`.

File: hedera-node/src/test/java/com/hedera/services/state/expiry/ExpiringCreationsTest.java [hunk 3]
```diff
@@ -224,6 +224,8 @@ void canOverrideTokenTransfers() {
 		assertEquals(someTokenXfers, actualRecord.asGrpc().getTokenTransferListsList());
 	}
 
+	// TODO can override hbar transfers
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/test/java/com/hedera/services/state/expiry/ExpiringCreationsTest.java:227]: nit: remove comment

File: hedera-node/src/test/java/com/hedera/services/state/logic/AwareNodeDiligenceScreenTest.java [hunk 4]
```diff
@@ -95,14 +98,15 @@
 
 	@BeforeEach
 	void setUp() {
-		subject = new AwareNodeDiligenceScreen(validator, txnCtx, backingAccounts);
+		// TODO:
```
Comment (Daniel-K-Ivanov) [inline hedera-node/src/test/java/com/hedera/services/state/logic/AwareNodeDiligenceScreenTest.java:101]: Is this leftover?

**Pull request closed without merging**